extmod/modframebuf.c

```diff
@@ -491,7 +491,11 @@ STATIC mp_obj_t framebuf_blit(size_t n_args, const mp_obj_t *args) {
     if (n_args > 4) {
         key = mp_obj_get_int(args[4]);
     }
-
+    mp_obj_framebuf_t *palette = MP_OBJ_NULL;
+    if (n_args > 5 && args[5] != mp_const_none) {
+        mp_obj_t palette_in = mp_obj_cast_to_native_base(args[5], MP_OBJ_FROM_PTR(&mp_type_framebuf));
+        palette = MP_OBJ_TO_PTR(palette_in);
```

Comment:
`mp_obj_cast_to_native_base` can return mp_const_none, but that's not the same as MP_OBJ_NULL.

So needs to be

```c
if (palette_in != mp_const_none) {
  palette = MP_OBJ_TO_PTR(palette_in);
}
```

in order to make the code below work.

Reply:
Sorry if I'm being dumb here but I'm afraid I don't understand this.

This check worked if I didn't pass `palette` or I explicitly passed a `palette` value `None`:
```C
if (n_args > 5 && args[5] != mp_const_none)
```
Are you saying that I need an additional check after the above, or that the above check should only be on `n_args` and be followed by this subsequent check? I'm not clear about the circumstances in which `palette_in` is `mp_const_none`.
```C
if (palette_in != mp_const_none) {
  palette = MP_OBJ_TO_PTR(palette_in);
}

Reply:
The case that this matters for is if `palette` is non-None but isn't derived from FrameBuffer.

But I said completely the wrong thing -- mp_obj_cast_to_native_base does actually return MP_OBJ_NULL in this case. Which means the code as you have it is exactly correct. Sorry!

Reply:
Ah! That's fortuitous as I hadn't considered that case :)